### PR TITLE
 Changes RetrieveRegistryCredentialsStep to use credential retrievers list.

### PR DESCRIPTION
**Pull request by coollog**
Part of #774

### CODE REVIEW

File: jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/RetrieveRegistryCredentialsStep.java [hunk 1]
```diff
@@ -21,42 +21,28 @@
...
-class RetrieveRegistryCredentialsStep implements AsyncStep<Authorization>, Callable<Authorization> {
+class RetrieveRegistryCredentialsStep implements AsyncStep<Credential>, Callable<Credential> {
 
   private static final String DESCRIPTION = "Retrieving registry credentials for %s";
 
-  /**
-   * Defines common credential helpers to use as defaults. Maps from registry suffix to credential
-   * helper suffix.
-   */
-  private static final ImmutableMap<String, String> COMMON_CREDENTIAL_HELPERS =
```
Comment (TadCordle) [inline jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/RetrieveRegistryCredentialsStep.java]: Maybe I'm just missing it, do we not look for these anymore?

Comment (coollog) [reply jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/RetrieveRegistryCredentialsStep.java]: These (and most of the code originally here) were moved to `CredentialRetrieverFactory`, where the default `CredentialRetriever`s are constructed.

File: jib-core/src/main/java/com/google/cloud/tools/jib/frontend/CredentialRetrieverFactory.java [hunk 2]
```diff
@@ -169,6 +170,9 @@ CredentialRetriever inferCredentialHelper(
               logger.info("  Caused by: " + ex.getCause().getMessage());
             }
           }
+
+        } catch (NonexistentServerUrlDockerCredentialHelperException | IOException ex) {
```
Comment (briandealwis) [inline jib-core/src/main/java/com/google/cloud/tools/jib/frontend/CredentialRetrieverFactory.java]: Should `NonexistentServerUrlDockerCredentialHelperException` and `NonexistentDockerCredentialHelperException ` be subclasses of `CredentialRetrievalException`?

Comment (coollog) [reply jib-core/src/main/java/com/google/cloud/tools/jib/frontend/CredentialRetrieverFactory.java]: That will potentially be done as part of #832

**Pull request merged**